tips/032_pii_masking.md

```diff
@@ -0,0 +1,242 @@
+## Effective AI Engineering #32: PII Masking
+
+**Are you sending customer emails and phone numbers to third-party AI models?** Every user query containing personal information gets transmitted to external services, creating compliance risks and privacy violations.
+
+Personal data in AI inputs creates liability for data breaches, regulatory violations, and user trust issues. But completely blocking PII often breaks legitimate use cases where personal context improves AI responses.
+
+### The Problem
+
+Many developers send user inputs directly to AI models without sanitizing personal information. This creates challenges that aren't immediately obvious:
+
+```python
+# BEFORE: Raw PII sent to AI models
+from mirascope.core import llm
+import lilypad
+
+@lilypad.trace()
+@llm.call(provider="openai", model="gpt-4o-mini")
+def handle_customer_support(customer_message: str) -> str:
+    return f"Help this customer with their request: {customer_message}"
+
+# Dangerous: PII gets sent to external AI service
+customer_inputs = [
+    "My email john.doe@company.com isn't receiving notifications",
+    "Call me at 555-123-4567 about my order #12345",
+    "My credit card 4532-1234-5678-9012 was charged twice",
+    "I live at 123 Main St, Boston MA 02101"
+]
+
+for message in customer_inputs:
+    # PII transmitted to AI provider
+    response = handle_customer_support(message)
+    print(f"Response: {response}")
+```
+
+**Why this approach falls short:**
+
+- **Privacy Violations:** Customer PII gets transmitted to third-party AI providers
+- **Compliance Risk:** GDPR, CCPA, and other regulations may be violated
+- **Data Retention:** AI providers may log and store customer personal information
+
+### The Solution: PII Detection and Placeholder Substitution
+
+A better approach is to detect PII, replace it with semantic placeholders, and restore the original values in responses. This pattern preserves context for AI processing while protecting sensitive data.
+
+```python
+# AFTER: PII masking with semantic placeholders
```

Comment:
this AFTER example should be super simple and I think it should use Presidio. This tutorial offers great insight to the basic thing we want to do: https://microsoft.github.io/presidio/samples/python/pseudonymization/

Again, we should make this STUPID simple.